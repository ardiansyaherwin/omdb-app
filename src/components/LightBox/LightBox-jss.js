const styles = (theme) => ({
  root: {
    margin: 0,
    padding: theme.spacing(2),
  },
  closeButton: {
    position: "absolute",
    right: theme.spacing(1),
    top: theme.spacing(1),
    color: 'white',
    background: "rgba(0,0,0,.9)",
  },
  dialogContent: {
    padding: "0 !important",
    border: 0,
  },
  dialogContentImage: {
    display: "block",
    width: "100%",
  },
});

export default styles;
