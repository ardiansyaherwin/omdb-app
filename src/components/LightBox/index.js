import React from "react";
import { PropTypes } from "prop-types";
import { withStyles } from "@material-ui/core/styles";
import {
  Dialog,
  DialogContent,
  IconButton,
} from "@material-ui/core";
import { Close } from "@material-ui/icons";

import styles from "./LightBox-jss";

const LightBox = (props) => {
  const { source, classes, showDialog, onClose } = props;
  const handleClose = () => {
    onClose()
  };

  return (
    <Dialog
      onClose={handleClose}
      aria-labelledby="customized-dialog-title"
      open={showDialog}
    >
      <DialogContent dividers className={classes.dialogContent}>
        <img src={source} className={classes.dialogContentImage} alt="" />
        <IconButton
          aria-label="close"
          className={classes.closeButton}
          onClick={handleClose}
        >
          <Close />
        </IconButton>
      </DialogContent>
    </Dialog>
  );
};

LightBox.propTypes = {
  source: PropTypes.string.isRequired,
  showDialog: PropTypes.bool.isRequired,
  onClose: PropTypes.func.isRequired,
};

export default withStyles(styles)(LightBox);
