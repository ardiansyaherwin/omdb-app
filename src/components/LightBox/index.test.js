import React from "react"
import { shallow } from "enzyme"
import LightBox from "./index"
import {
  InputBase,
  CircularProgress,
  IconButton,
  FormHelperText,
} from "@material-ui/core"
import { unwrap } from "@material-ui/core/test-utils"

const LightBoxNaked = unwrap(LightBox)

const requiredProps = {
  source: "http://image.com/image.jpg",
  showDialog: false,
  onClose: jest.fn(),
  classes: {}
}

describe("<LightBox />", () => {

  it("should render as expected", () => {
    const wrapper = shallow(<LightBoxNaked {...requiredProps} />);
    expect(wrapper).toMatchSnapshot()
  })

  it("should have props:poster value which suffixed by .jpg, .jpeg or .png", () => {
    const isValidImage = requiredProps.source.match(/\.(jpeg|jpg|)$/) != null;
    expect(isValidImage).toBeTruthy();
  });

  it("expect to call props:onClose if <IconButton /> is clicked", () => {
    const wrapper = shallow(<LightBoxNaked {...requiredProps} />)
    wrapper
      .find(IconButton)
      .simulate("click")

    wrapper.update()
    expect(requiredProps.onClose).toHaveBeenCalledTimes(1)
  })

})
