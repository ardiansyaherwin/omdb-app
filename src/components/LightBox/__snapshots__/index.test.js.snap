// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LightBox /> should render as expected 1`] = `
<WithStyles(ForwardRef(Dialog))
  aria-labelledby="customized-dialog-title"
  onClose={[Function]}
  open={false}
>
  <WithStyles(ForwardRef(DialogContent))
    dividers={true}
  >
    <img
      alt=""
      src="http://image.com/image.jpg"
    />
    <WithStyles(ForwardRef(IconButton))
      aria-label="close"
      onClick={[Function]}
    >
      <CloseIcon />
    </WithStyles(ForwardRef(IconButton))>
  </WithStyles(ForwardRef(DialogContent))>
</WithStyles(ForwardRef(Dialog))>
`;
